### ISSUE TITLE
World caching problems

### ISSUE THREAD

**Issue by etiaro** [rob9315/mcproxy-old#14]
Hi,
I'm experiencing bugs with chunks caching: 
blank textures: I found this one pretty interesting, as client knows where are blocks(and their types), just refuses to show textures for some reason. Sometimes enabling and disabling x-ray shows the textures, but when go to neighbour chunk and look away, the corrupted chunk becomes blank again. Also, mods like chest-ESP are showing positions correctly

Shifting chunks: on 2b2t's spawn, or the nether(so probably on chunks filled with much blocks), some of the chunks are getting shifted down. Seems like always y is subtracted by multiply of some number(I didn't count tbh). It's best visible in the nether, as the roof is flat bedrock. Almost all of broken chunk's roofs are on the same level, and a few looks like they've got shifted twice. Sometimes there's just one corrupted chunk, and sometimes there is whole line(2x8 was largest i saw), in some cases they're completely random, I can't see any pattern there.

I am running proxy on really low-cpu and low-RAM(1gb ram raspberry pi, arch linux) so maybe it somehow causes the problem.

**Comment by rob9315** [comment]
>blank textures: I found this one pretty interesting, as client knows where are blocks(and their types), just refuses to show textures for some reason. Sometimes enabling and disabling x-ray shows the textures, but when go to neighbour chunk and look away, the corrupted chunk becomes blank again. Also, mods like chest-ESP are showing positions correctly

Ah yes, I know this one. This happens because of the way [mineflayer](https://github.com/PrismarineJS/mineflayer) calculates/caches light information, which gets out of sync whenever you change something and relog.

>Shifting chunks: on 2b2t's spawn, or the nether(so probably on chunks filled with much blocks), some of the chunks are getting shifted down. Seems like always y is subtracted by multiply of some number(I didn't count tbh). It's best visible in the nether, as the roof is flat bedrock. Almost all of broken chunk's roofs are on the same level, and a few looks like they've got shifted twice. Sometimes there's just one corrupted chunk, and sometimes there is whole line(2x8 was largest i saw), in some cases they're completely random, I can't see any pattern there.

I have no clue though, what causes this.

>I am running proxy on really low-cpu and low-RAM(1gb ram raspberry pi, arch linux) so maybe it somehow causes the problem.

I don't think that has anything to do with it, but good that you use arch BTW :1st_place_medal: 

TLDR: The lighting issue can be fixed, the PrismarineJS team was working on that already at some point, breaking a block should also fix it. The chunk shifting problem i have no clue why it happens or how to fix it, sorry.

**Comment by MrGeorgen** [comment]
related issues to the lighting problem:
https://github.com/PrismarineJS/mineflayer/issues/591
https://github.com/PrismarineJS/prismarine-world/issues/46
